connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -403,7 +403,16 @@ private static ConfigInfos validateClientOverrides(String connName,
         List<ConfigInfo> configInfoList = new LinkedList<>();
         Map<String, ConfigKey> configKeys = configDef.configKeys();
         Set<String> groups = new LinkedHashSet<>();
-        Map<String, Object> clientConfigs = connectorConfig.originalsWithPrefix(prefix);
+        Map<String, Object> clientConfigs = new HashMap<>();
+        for (Map.Entry<String, Object> rawClientConfig : connectorConfig.originalsWithPrefix(prefix).entrySet()) {
+            String configName = rawClientConfig.getKey();
+            Object rawConfigValue = rawClientConfig.getValue();
+            ConfigKey configKey = configDef.configKeys().get(configName);
+            Object parsedConfigValue = configKey != null
+                ? ConfigDef.parseType(configName, rawConfigValue, configKey.type)
```

Comment:
Would it be a lot easier to do the following?
```
Map<String, Object> prefixedOriginals = connectorConfig.originalsWithPrefix(prefix);
Map<String, Object> clientConfigs = configDef.parse(prefixedOriginals);
```
This does check a few more things (dependencies are set and validators are run), so maybe that's not really want we want to do here.

Reply:
@rhauch yeah, that was my first pass. Unfortunately, `ConfigDef.parse(...)` throws an exception if required configs are missing, and since this is only the set of connector-specific overrides and not the complete client config, there's no guarantee that, e.g., `bootstrap.servers` will be included.